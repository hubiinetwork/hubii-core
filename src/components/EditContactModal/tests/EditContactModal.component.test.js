// import React from 'react';
// import { shallow, mount, dive } from 'enzyme';
// import EditContactModals, { EditContactModal } from '../EditContactModal.component';
// import { StyledButton1, ParentDiv } from '../EditContactModal.style';
// // Import Theme Provider
// import { ThemeProvider } from 'styled-components';
// // Import Themes
// import dark from 'themes/darkTheme';


// describe('<EditContactModal/>', () => {
//   const props = {
//     name: 'mike',
//     address: '0x12312313',
//     onEdit: jest.fn(),
//     onChange: jest.fn(),
//     validateEdit: jest.fn(),
//   };
//   it('should correctly render <EditContactModal/>', () => {
//     const wrapper = shallow(
//       <EditContactModals {...props} />
//       );
//     expect(wrapper).toMatchSnapshot();
//   });

//   // not completely sure how to test this
//   describe('<EditContactModal/> function testing', () => {
//     it('on submitting the form the handleEdit function should run correctly', () => {
//       const wrapper = shallow(
//         <EditContactModals {...props} />

//         );
//       const spy = jest.spyOn(EditContactModal.prototype, 'handleEdit');
//       const button = wrapper.find(EditContactModal).dive().find(StyledButton1);
//       button.simulate('click');
//       console.log(spy);
//       expect(spy).toHaveBeenCalledTimes(1);
//       // expect(wrapper.find(EditContactModal).dive().props.onEdit).toHaveBeenCalledTimes(1);
//     });

//     it('should run validateEdit function', () => {
//       const wrapper = shallow(
//         <EditContactModals {...props} />
//       );
//       const spy = wrapper.find(EditContactModal);
//       console.log(spy);
//       // spy.validateEdit({}, 'hello', 'yo');
//       // expect(validateEdit({}, 'hello', 'yo')).toEqual(true);
//     });
//   });
// });
