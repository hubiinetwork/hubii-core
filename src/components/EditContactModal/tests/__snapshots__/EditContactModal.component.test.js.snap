// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditContactModal/> should correctly render <EditContactModal/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(EditContactModal)
    address="0x12312313"
    name="mike"
    onChange={[Function]}
    onEdit={[Function]}
    validateEdit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "address": "0x12312313",
      "form": Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      },
      "name": "mike",
      "onChange": [Function],
      "onEdit": [Function],
      "validateEdit": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "address": "0x12312313",
        "form": Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        },
        "name": "mike",
        "onChange": [Function],
        "onEdit": [Function],
        "validateEdit": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
